source/dub/generators/build.d

```diff
@@ -99,18 +100,44 @@ class BuildGenerator : ProjectGenerator {
 			foreach (dep; ti.dependencies)
 				buildTargetRec(dep);
 
-			NativePath[] additional_dep_files;
+			NativePath[] additional_dep_files, dependencyBinariesToCopy;
 			auto bs = ti.buildSettings.dup;
 			foreach (ldep; ti.linkDependencies) {
-				if (bs.targetType != TargetType.staticLibrary && !(bs.options & BuildOption.syntaxOnly)) {
-					bs.addSourceFiles(target_paths[ldep].toNativeString());
+				auto location = target_paths[ldep].toNativeString();
+
+				if (bs.targetType != TargetType.staticLibrary && !(bs.options & BuildOption.syntaxOnly) && isLinkerFile(settings.platform, location)) {
```

Comment:
where does this `isLinkerFile(settings.platform, location)` come from? doesn't seem related to the change and I don't quite see how this makes sense here either.

Reply:
It comes from ``dub.compilers.utils``.

```d
bool isLinkerFile(const scope ref BuildPlatform platform, string f)
{
	import std.path;
	switch (extension(f)) {
		default:
			return false;
		case ".lib", ".obj", ".res", ".def":
			return platform.platform.canFind("windows");
		case ".a", ".o", ".so", ".dylib":
			return !platform.platform.canFind("windows");
	}
}
```

Basically it prevents ``.dll`` files from being added as a source file (which errors). This is the original error that triggered me to write this PR.